app/forms/flood_risk_engine/steps/base_address_form.rb

```diff
@@ -0,0 +1,131 @@
+module FloodRiskEngine
+  module Steps
+    class BaseAddressForm < BaseRedirectableForm
+
+      property :uprn
+      property :address_list, virtual: true
+
+      validates :uprn, presence: { message: I18n.t("flood_risk_engine.validation_errors.uprn.blank") }
+
+      validates :uprn, numericality: {
+        only_integer: true,
+        allow_blank: true,
+        message: I18n.t("flood_risk_engine.validation_errors.uprn.format")
+      }
+
+      # Post code is set in previous page, and must be passed in so we can create the
+      # address drop down list - need this externally set as we can't access Session in the Form
+      attr_reader :post_code
```

Comment:
Yes.  IMHO until they choose the address the association remains empty so its clear no address has been selected. I do not want to create an address with only a postcode, firstly that is simply not the address, then we have  abondonment, back button, back link, change postcode aand check details to handle .. and we are suddenly managing weird edge cases